plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/decoder/EmptyFieldHandlingAvroRowDecoder.java

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka.decoder;
+
+import io.trino.decoder.DecoderColumnHandle;
+import io.trino.decoder.FieldValueProvider;
+import io.trino.decoder.RowDecoder;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.Type;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.plugin.kafka.schema.confluent.AvroSchemaConverter.DUMMY_FIELD_NAME;
+import static io.trino.plugin.kafka.schema.confluent.AvroSchemaConverter.DUMMY_ROW_TYPE;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toUnmodifiableMap;
+
+public class EmptyFieldHandlingAvroRowDecoder
+        implements RowDecoder
+{
+    public static final String NAME = "avro";
+
+    private static final FieldValueProvider NULL_FIELD_VALUE_PROVIDER = new FieldValueProvider()
+    {
+        private final Block nullBlock = getNullBlock();
+
+        private Block getNullBlock()
+        {
+            BlockBuilder blockBuilder = DUMMY_ROW_TYPE.createBlockBuilder(null, 1);
+            BlockBuilder singleRowBuilder = blockBuilder.beginBlockEntry();
+            singleRowBuilder.appendNull();
+            blockBuilder.closeEntry();
+            return blockBuilder.getObject(0, Block.class);
+        }
+
+        @Override
+        public Block getBlock()
+        {
+            return nullBlock;
+        }
+
+        @Override
+        public boolean isNull()
+        {
+            return false;
+        }
+    };
+    private final RowDecoder delegate;
+
+    public EmptyFieldHandlingAvroRowDecoder(RowDecoder delegate)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+    }
+
+    @Override
+    public Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeRow(byte[] data)
+    {
+        return delegate.decodeRow(data).map(decodedRow -> decodedRow.entrySet().stream()
+                .map(entry -> {
+                    if (isDummyRowType(entry.getKey().getType())) {
+                        return new AbstractMap.SimpleImmutableEntry<>(entry.getKey(), NULL_FIELD_VALUE_PROVIDER);
+                    }
+                    else {
+                        return entry;
+                    }
+                })
+                .collect(toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue)));
```

Comment:
some reason to not use Guava map?

Reply:
changed everything to guava.

Reply:
yes, also commented below (for lists): maps may contain non-null key with null values. Also added a comment in the code to explain.